OpenRA.Mods.Common/Traits/Attack/AttackFollow.cs

```diff
@@ -343,6 +356,39 @@ public override bool Tick(Actor self)
 				if (useLastVisibleTarget && !lastVisibleTarget.IsValidFor(self))
 					return true;
 
+				// If all valid weapons have depleted their ammo and Rearmable trait exists, return to RearmActor to reload
+				// and resume the activity after reloading if AbortOnResupply is set to 'false'
+				if (rearmable != null && !useLastVisibleTarget && attack.Armaments.All(x => x.IsTraitPaused || !x.Weapon.IsValidAgainst(target, self.World, self)))
+				{
+					// Attack moves never resupply
+					if (source == AttackSource.AttackMove)
+						return true;
+
+					// AbortOnResupply cancels the current activity (after resupplying) plus any queued activities
+					if (attack.Info.AbortOnResupply)
+						NextActivity?.Cancel(self);
+
+					if (isAircraft)
+						QueueChild(new ReturnToBase(self));
+					else
+					{
+						if (attack.Info.AbortOnResupply)
```

Comment:
Why is it doing this just if we are aborting? It should do the same as for aircraft: Resupply and act according to `AbortOnResupply` only after that.

Reply:
That does make sense